src/test/java/ServletTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.logging.Logger;
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.common.base.Charsets;
+import com.google.common.hash.Hashing;
+import com.google.googleidentity.filter.UserAuthenticationFilter;
+import com.google.googleidentity.security.LoginCheckServlet;
+import com.google.googleidentity.security.UserSession;
+import com.google.googleidentity.user.InMemoryUserDetailsService;
+import com.google.googleidentity.user.UserDetails;
+import com.google.googleidentity.user.UserDetailsService;
+import com.google.inject.Injector;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+import com.google.inject.Provider;
+
+import org.junit.Before;
+import org.junit.Test;
+
+
+import static org.mockito.Mockito.*;
+
+public class ServletTest  {
```

Comment:
Normally unit tests shall test a single class, and let's put tests under the same package in tests folder, for example, if the test is for src/main/java/com/google/googleidentity/security/LoginServlet.java file, then let's put the test under src/test/java/com/google/googleidentity/security and naming it as LoginServletTest.java.